enhancements/cluster-logging/cluster-logging-move-init-to-es-operator.md

```diff
@@ -0,0 +1,80 @@
+---
+title: cluster-logging-log-forwarding
+authors:
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@ewolinetz"
+reviewers:
+  - "@bparees"
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@alanconway"
+  - "@ewolinetz"
+approvers:
+  - "@ewolinetz"
+  - "@jcantrill"
+creation-date: 2019-09-17
+last-updated: 2019-10-25
+status: implementable
+see-also:[]
+replaces:[]
+superseded-by:[]
+---
+
+# cluster-logging-es-init
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of moving Elasticsearch initialization logic to the elasticsearch-operator is to reduce the number of times these init scripts executed.
+At the moment we keep these scripts in the ES image, and execute them everytime when ES pod starts. 
+
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide an API that allows users to specify init procedures for the ES cluster
+* Allow users to choose what command has to be executed during the init process for NEW and UPGRADE elasticsearch cluster states
+
+## Proposal
+
+The following changes will define how the elasticsearch-operator executes init process for Elasticsearch cluster fresh start and upgrade
+
+* Scripts will be executed from outside the ES image
```

Comment:
Maybe should say something like: Initialization logic will be executed from outside the ES container

Reply:
+1 i thought we make mention of running in a separate pod though somewhere...

Reply:
We should be specific here for future readers. Do we spin a separate pod? an initContainer? a container in a multi-container pod?

Reply:
we reference it here: https://github.com/openshift/enhancements/pull/232/files#diff-ffaf040ed47f889b90a1007c1822474dR73 

It will be an external run object -- but we should echo that up here.

Reply:
Fixed. Added additional info about the way we going to launch these scripts.